### PR TITLE
[staging] cdparanoia: cleanup, fix build on powerpc

### PR DESCRIPTION
**Pull request by r-burns**
The configure helpers are old and already need to be updated on aarch64,
so let's just do this on all non-x86 platforms. This will probably fix
other architectures that weren't well-supported at the time.

###### Motivation for this change

Fixes build on ppc64le (and possibly others)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JohnAZoidberg** [comment]
Does it build on powerpc like this?
I also see

```
  nativeBuildInputs = lib.optional stdenv.isAarch64 autoreconfHook;
```

should this also be nonx86 then?

**Comment by r-burns** [comment]
Huh, I didn't notice that bit, and it is indeed not needed on powerpc, as the derivation builds for me as is. Interestingly, I'm able to remove the autoreconfHook on aarch64 and it still builds there as well. I think updating the configure.{sub,guess} scripts is sufficient, because they are only needed at configure time.
It looks like @dezgeg added this in https://github.com/NixOS/nixpkgs/commit/c863aa384e1a6237380afffd0feaf28b48983a23 - maybe there's something I'm missing, or it's no longer needed. Judging by the commit message, he may have just been fed up after wrestling with the build ;)

**Comment by JohnAZoidberg** [comment]
> Interestingly, I'm able to remove the autoreconfHook on aarch64 and it still builds there as well.

Cook, thanks for checking! Let's remove it then.

**Comment by r-burns** [comment]
Ok, done! And just to be sure, I also checked x86 -> aarch64, x86 -> powerpc, and powerpc -> x86 cross.

**Comment by r-burns** [comment]
Head commit was working but the merge commit was failing on borg aarch64 because the configure snippet was being appended as args to `unset`. This is now a mass rebuild, so I'll target staging (after the next automatic master -> staging merge, so I don't cause a mass notification).

**Comment by JohnAZoidberg** [comment]
> Ok, done! And just to be sure, I also checked x86 -> aarch64, x86 -> powerpc, and powerpc -> x86 cross.

Amazing, thanks!

> Head commit was working but the merge commit was failing on borg aarch64 because the configure snippet was being appended as args to unset. This is now a mass rebuild, so I'll target staging (after the next automatic master -> staging merge, so I don't cause a mass notification).

Hmm, I don't quite understand. Why does it cause a mass rebuild now, but not before?

**Comment by r-burns** [comment]
Haha I was super confused for a minute but now I see what the issue was. See the semicolon at the end of this line?
https://github.com/NixOS/nixpkgs/blob/ede9879e64b9cf477c5f1677601421d2d4dc1bf2/pkgs/applications/audio/cdparanoia/default.nix#L32
That actually goes into the shell, and is what keeps the first `cp` command from being appended as an arg to `unset`. Somewhere along the line I lost that semicolon and ofborg started complaining.

I kind of want to get rid of this weird trick and just put the `unset` command in its own multiline string, although doing so does cause a mass rebuild. Up to you I guess.

**Review by JohnAZoidberg — APPROVED** [review]
Ohhh, I see what's happening here. cdparanoia is a transitive dependency of all of those packages.
Before your change the nix expression would evaluate to the same derivation but now it doesn't anymore.

Yeah that semicolon is definitely not obvious, I think it's good to change that.

**Comment by r-burns** [comment]
@JohnAZoidberg I see you are a committer, do you think this is ready for merge? Or shall I solicit for further review?

**Comment by zhaofengli** [comment]
Can confirm that this fixes the build on riscv64 as well. Let's get this merged.

**Pull request merged**
